include/coap3/coap_oscore.h

```diff
@@ -193,6 +198,45 @@ int coap_new_oscore_recipient(coap_context_t *context,
 int coap_delete_oscore_recipient(coap_context_t *context,
                                  coap_bin_const_t *recipient_id);
 
+/**
+ * Opaque pointer for internal oscore_ctx_t type.
+ */
+typedef struct oscore_ctx_t * oscore_ctx_handle_t;
+
+/**
+ * Opaque pointer for internal oscore_recipient_ctx_t type.
+ * 
+ */
+typedef struct oscore_recipient_ctx_t * oscore_recipient_ctx_handle_t;
```

Comment:
In general for libcoap, all the other` typedef struct X X;` statements do not have the `*` (pointer) included.  We need to be consistent here. 

In general, any forward declarations should be included in the appropriate place in `include/coap3/coap_forward_decls.h`.

As we are trying to keep the public API to have everything prefixed with `coap_` to prevent any name clashes, I think perhaps the 2 forward references (in `include/coap3/coap_forward_decls.h`) should be
````
typedef struct oscore_ctx_t coap_oscore_handle_t
typedef struct oscore_recipient_ctx_t coap_oscore_recipient_ctx_handle_t
````
and then referenced in your changes with the new names.